pulp_container/app/access_policy.py

```diff
@@ -0,0 +1,30 @@
+from pulpcore.plugin.access_policy import AccessPolicyFromDB
+
+from pulp_container.app import models
+
+
+class DistributionAccessPolicyFromDB(AccessPolicyFromDB):
+    """
+    Access policy for DistributionViewSet which handles namespace permissions.
+    """
+
+    def has_add_namespace_perms(self, request, view, action):
+        """
+        Check whether a user can create a namespace or can read the namespace if it already exists.
+        """
+        namespace = request.data["base_path"].split("/")[0]
+        try:
+            namespace = models.ContainerNamespace.objects.get(name=namespace)
+        except models.ContainerNamespace.DoesNotExist:
+            return request.user.has_perm("container.add_containernamespace")
+        else:
+            return request.user.has_perm(
```

Comment:
i guess i would then replace this with 'container.manipulate_distributions_namespace' This permission will be define on the namespace